sdk/src/logs/logger.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/logger.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+Logger::Logger() noexcept : processor_{nullptr} {}
+
+Logger::Logger(std::shared_ptr<LogProcessor> processor) noexcept : processor_{processor} {}
+
+void Logger::log(const opentelemetry::logs::LogRecord &record) noexcept
+{
+  // TODO: Sampler
+  // If this log's severity is less than the minimum severity, do nothing
+  if (!IsEnabled(record.severity))
+    return;
+
+  /**
+   * Convert the LogRecord to the heap first before sending to processor.
+   * TODO: Change the API log(LogRecord) function to log(*LogRecord) so the following line
+   * converting record a heap variable can be removed
+   */
+  auto record_pointer =
+      std::unique_ptr<opentelemetry::logs::LogRecord>(new opentelemetry::logs::LogRecord(record));
+
+  // Inject timestamp
+  record_pointer->timestamp = core::SystemTimestamp(std::chrono::system_clock::now());
+
+  // TODO: inject traceid/spanid later
+
+  // Send the log record to the processor
+  GetProcessor()->OnReceive(std::move(record_pointer));
```

Comment:
Can this lead to crash/seg-fault if Processor was not initialized while creating `LoggerProvider` instance ? Should we check for nullptr before using here or at `LoggerProvider` level.

Reply:
Yes, will add a nullptr check